elasticsearch/file/src/test/java/nl/knaw/huc/resources/FileResourcePageTurnerTest.java

```diff
@@ -0,0 +1,230 @@
+import org.mockito.MockitoAnnotations;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpRequest;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.UUID;
+
+import static javax.ws.rs.client.Entity.entity;
+import static nl.knaw.huc.TestUtils.getResourceAsBytes;
+import static nl.knaw.huc.service.JsonPathFactory.withJackson;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.verify.VerificationTimes.once;
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class FileResourcePageTurnerTest {
+
+  private static ClientAndServer mockServer;
+  private static final int mockPort = 1080;
+
+  private static final File configFile = new File("src/test/resources/test-config-pagesize-2.yml");
+
+  public static DropwizardAppExtension<FileConfiguration> application;
+
+  /*
+    To test logging: create test app with DocumentResource and custom config and file logger
+   */
+  static {
+    try {
+      var factory = new YamlConfigurationFactory<>(
+          FileConfiguration.class,
+          Validators.newValidator(),
+          Jackson.newObjectMapper(),
+          "dw"
+      );
+
+      var fileConfiguration = factory.build(configFile);
+      application = new DropwizardAppExtension<>(FileIndexer.class, fileConfiguration);
+
+    } catch (IOException | ConfigurationException ex) {
+      throw new RuntimeException("Could not init test app", ex);
+    }
+  }
+
+  @BeforeAll
+  public static void setUpClass() {
+    mockServer = ClientAndServer.startClientAndServer(mockPort);
+  }
+
+  @BeforeEach
+  public void before() {
+    mockServer.reset();
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    mockServer.stop();
+  }
+
+  @Test
+  public void testFields_retrievesAllVersionPages() throws IOException {
+    // start resource mocks:
+    var mockRequests = startTextrepoMockServer();
+    // start version page resource mocks:
+    var versionsEndpoint = "/rest/files/[a-f0-9-]*/versions";
+    mockRequests.add(mockEndpointPage(versionsEndpoint, "textrepo-versions-page1.json", 0, 2));
+    mockRequests.add(mockEndpointPage(versionsEndpoint, "textrepo-versions-page2.json", 2, 2));
+    mockRequests.add(mockEndpointPage(versionsEndpoint, "textrepo-versions-page3.json", 4, 2));
+    // create test file:
+    var fileContents = getResourceAsBytes("file.txt");
+    var fileId = UUID.randomUUID().toString();
+
+    var response = postTestContents(fileContents, "text/plain", fileId);
+    var fields = response.readEntity(String.class);
+
+    assertThat(response.getStatus()).isEqualTo(200);
+    assertThat(JsonPath.parse(fields).read("$.file.id", String.class)).isEqualTo(fileId);
```

Comment:
Kun je hier JsonPath.parse(fields) nog buiten de haakjes halen (refactor > extract variable), of werkt die read(..) maar 1x na parse()?
Zou de volgende asserts iets makkelijker te lezen maken, als het wel kan.

Reply:
tsjakka!
3051387ef7389c8436ea7f486749af1a083c9fbd